File: javaimpl/src/main/scala/com/jw/sharepoint/examples/SharepointAuthenticator.scala
package com.jw.sharepoint.examples

import java.net.Authenticator
import java.net.PasswordAuthentication
import java.util.Properties

class SharepointAuthenticator(properties: Properties) extends Authenticator {
  override def getPasswordAuthentication: PasswordAuthentication = new PasswordAuthentication(properties.getProperty("username"), properties.getProperty("password").toCharArray)
}

File: javaimpl2/src/main/java/com/jw/sharepoint/examples/SharepointAuthenticator.java
package com.jw.sharepoint.examples;

import java.net.Authenticator;
import java.net.PasswordAuthentication;
import java.util.Properties;

public class SharepointAuthenticator extends Authenticator{
    
	private Properties properties;
	
	public SharepointAuthenticator(Properties props){
		properties = props;
	}
    
    public PasswordAuthentication getPasswordAuthentication () {
	
	    return new PasswordAuthentication (        	
        	properties.getProperty("username"),properties.getProperty("password").toCharArray());
	}


}
